keyboards/preonic/keymaps/smt/keymap.c

```diff
@@ -0,0 +1,248 @@
+#include "preonic.h"
+#include "action_layer.h"
+#include "eeconfig.h"
+#ifdef AUDIO_ENABLE
+  #include "audio.h"
+#endif
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+#define _DVORAK 0
+#define _QWERTY 1
+#define _LOWER 2
+#define _RAISE 3
+#define _ADJUST 16
+
+enum preonic_keycodes {
+  DVORAK = SAFE_RANGE,
+  QWERTY,
+  COLEMAK,
```

Comment:
If I remove COLEMAK, the build and flash seem to go ok, but the keyboard is unresponsive after starting up. Once that happened, I built the default image, flashed, and incrementally added my changes back in until I narrowed it down to this line.

It is in a working state like this, for now, and when I have a moment later today, I'll continue to play with it. It's an opportunity to introduce myself to the debugging procedure, but I have never done anything with C before now 😄 

For context, I have been using your docker image to build, and my workflow is pretty much this:

```
$ rm ./preonic_smt.hex
$ docker run \
  -e keymap=smt -e keyboard=preonic -e subproject='' \
  --rm -v $(pwd):/qmk:rw jackhumbert/qmk_firmware make clean
$ docker run \
  -e keymap=smt -e keyboard=preonic -e subproject='' \
  --rm -v $(pwd):/qmk:rw jackhumbert/qmk_firmware make
<put keyboard into flash mode>
$ dfu-programmer atmega32u4 erase
$ dfu-programmer atmega32u4 flash ./preonic_smt.hex
$ dfu-programmer atmega32u4 start
```